app/services/course/assessment/question/codaveri_problem_generation_service.rb

```diff
@@ -26,28 +26,53 @@ def codaveri_generate_problem
   # Creates a new service codaveri feedback rating object.
   #
   # @param [Course::Discussion::Post::CodaveriFeedback] feedback Feedback to be sent to Codaveri
-  def initialize(assessment, custom_prompt, language, version, difficulty)
+  def initialize(assessment, params, language, version)
+    custom_prompt = params[:custom_prompt]
     @payload = {
       userId: assessment.creator_id.to_s,
       courseName: assessment.course.title,
-      config: {
-        difficulty: difficulty,
-        customPrompt: custom_prompt.length >= 500 ? "#{custom_prompt[0...495]}..." : custom_prompt,
-        testcasesType: 'expression',
-        languageVersion: {
-          language: language,
-          version: version
-        }
+      languageVersion: {
+        language: language,
+        version: version
+      },
+      llmConfig: {
+        customPrompt: (custom_prompt.length >= 500) ? "#{custom_prompt[0...495]}..." : custom_prompt,
+        testcasesType: 'expression'
       },
       requireToken: true,
       tokenConfig: {
         returnResult: true
       }
     }
+
+    return unless params[:is_first_generated_question] == 'false'
```

Comment:
Nit: frontend use camelCase

Reply:
this one is actually backend use case. we only pass by the variable from frontend to backend and hence the reason this uses snake_case (and so as the other attributes, for example `public_test_cases`)